forte/data/data_store.py

```diff
@@ -302,6 +302,7 @@ def __getstate__(self):
         state.pop("_DataStore__tid_ref_dict")
         state.pop("_DataStore__tid_idx_dict")
         state.pop("_DataStore__deletion_count")
+        state.pop("_type_attributes", None)
```

Comment:
I think it might be useful to mention the instance when the data store state has the `_type_attributes` key and why we need to remove it.

Reply:
Explanations added to the docstring